### ISSUE TITLE
Choppy audio (both music and sound) in 3.0.1 on WinXP SP3

### ISSUE THREAD

**Issue by earwigsinyourveins** [chocolate-doom/chocolate-doom#1311]
Audio in the latest release of Chocolate Doom lags on Windows XP SP3 with all latest updates installed. Tested in both Chocolate Doom and Chocolate Heretic.
Example (link is limited to 3 days): https://litter.catbox.moe/oxmd18.wav

**Comment by vorob1** [comment]
Same here. Wanted to use CH to play Heretic on old laptop and sound and music are messed up...

**Comment by Bluethunder83** [comment]
I have an older system that I use for Windows XP.  The music is fine but the sound fx's are choppy sounding.  The same for Crispy as well obviously.

**Comment by fabiangreffrath** [comment]
Guys, Windows XP has been obsoleted by three generations of operating systems now.

**Comment by Bluethunder83** [comment]
Indeed it has and It has no significance but I thought it was interesting that chocolate and crispy do this in xp but Woof it doesn't

**Comment by vorob1** [comment]
You are absolutely right. But in this case windows xp should me mentioned in unsupported section like windows 9x does.

I’ve got old laptop with Yamaha chip that supports general midi. I wanna play heretic as native as possible and crispy doom is a perfect way to do this. It will show game with better screen resolution and use my hardware stuff to play music, not emulate it. But crispy doesn’t work at all, so I had to switch to chocolate one, and not the latest version but 3.0.0. This one works good.

**Comment by RamonUnch** [comment]
SDL2 works fine under XP, 
However choco-doom builds SDL without DirectSound support and the WINMM api is used instead. This later dates back from Win3.1 and is very unresponsive, you need to increase some buffer from 28ms to 48ms (solution 1).

1) set `snd_maxslicetime_ms           48` in chocolate-doom.cfg

2) Replace SDL2.DLL with another build, I make my own builds but the official one works fine:
https://www.libsdl.org/download-2.0.php
Just download the Win32 version and extract in choco-doom folder replacing the old one.
In this case you can set `snd_maxslicetime_ms           28` or even 24 or even maybe lower depending on your config.
This second solution is preferred.

**Comment by vorob1** [comment]
RamonUnch thank you very much. While replacing SDL2.DLL didn't help, changing snd_maxslicetime_ms to 48 made my life better, now "farting" sounds are less common in the game. I can even play Crispy doom / heretic now.

**Comment by Bluethunder83** [comment]
set snd_maxslicetime_ms 48 in chocolate-doom.cfg

This worked perfect for me.  Fixed the crackling and choppy sound effects. Thanks

**Comment by vorob1** [comment]
Well, i can say that setting to 48 solves the issue, but not completely. I'm still getting some "clicking" in sound effects.

**Comment by RamonUnch** [comment]
Maybe try a combo of latest SDL and set the environment variable
`SDL_AUDIODRIVER=directsound`
See https://github.com/fabiangreffrath/crispy-doom/issues/718#issuecomment-827845141 for more details.
Maybe try to use the old SDL2.07 that still supported XAudio2.
